bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,414 @@
+
+    class_template = parsed_info["members"][0]
+
+    assert class_template["kind"] == "CLASS_TEMPLATE"
+    assert class_template["name"] == "AStruct"
+
+    template_non_type_parameter = class_template["members"][0]
+
+    assert template_non_type_parameter["kind"] == "TEMPLATE_NON_TYPE_PARAMETER"
+    assert template_non_type_parameter["element_type"] == "Int"
+    assert template_non_type_parameter["name"] == "N"
+
+
+def test_function_template(tmp_path):
+    file_contents = """
+    template <typename T>
+    void aFunction() {}
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    function_template = parsed_info["members"][0]
+
+    assert function_template["kind"] == "FUNCTION_TEMPLATE"
+    assert function_template["result_type"] == "void"
+    assert function_template["name"] == "aFunction"
+
+    template_type_parameter = function_template["members"][0]
+
+    assert template_type_parameter["kind"] == "TEMPLATE_TYPE_PARAMETER"
+    assert template_type_parameter["name"] == "T"
+    assert template_type_parameter["access_specifier"] == "PUBLIC"
+
+
+def test_template_type_parameter(tmp_path):
+    file_contents = """
+    template <typename T>
+    struct AStruct {};
+
+    template <typename P>
+    void aFunction() {}
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    class_template = parsed_info["members"][0]
+    template_type_parameter = class_template["members"][0]
+
+    assert template_type_parameter["kind"] == "TEMPLATE_TYPE_PARAMETER"
+    assert template_type_parameter["element_type"] == "Unexposed"
+    assert template_type_parameter["name"] == "T"
+
+    function_template = parsed_info["members"][1]
+    template_type_parameter = function_template["members"][0]
+
+    assert template_type_parameter["kind"] == "TEMPLATE_TYPE_PARAMETER"
+    assert template_type_parameter["element_type"] == "Unexposed"
+    assert template_type_parameter["name"] == "P"
+
+
+def test_default_delete_constructor(tmp_path):
+    file_contents = """
+    class aClass {
+        aClass() = default;
+
+        // disable the copy constructor
+        aClass(double) = delete;
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    class_decl = parsed_info["members"][0]
+
+    default_constructor = class_decl["members"][0]
+
+    assert default_constructor["kind"] == "CONSTRUCTOR"
+    assert default_constructor["name"] == "aClass"
+    assert default_constructor["result_type"] == "void"
+    assert default_constructor["is_default_constructor"]
+
+    delete_constructor = class_decl["members"][1]
+
+    assert delete_constructor["kind"] == "CONSTRUCTOR"
+    assert delete_constructor["name"] == "aClass"
+    assert delete_constructor["result_type"] == "void"
+    # no check available for deleted ctor analogous to `is_default_constructor`
```

Comment:
TODO:

Add test for template instantiation parsing since that's bread and butter for PCL bindings